packages/terminal/src/browser/terminal-widget.ts

```diff
@@ -228,29 +236,79 @@ export class TerminalWidget extends BaseWidget implements StatefulWidget {
             }
             return;
         }
+        this.started();
+    }
 
+    protected async openTerm(): Promise<void> {
+        this.isOpeningTerm = true;
+
+        if (this.isTermOpen === true) {
+            this.isOpeningTerm = false;
+            return Promise.reject("Already open");
+        }
+
+        /* Wait for the backend terminal to be requested before opening the xterm terminal.  */
+        await this.waitForStarted;
+
+        if (this.terminalId === undefined) {
+            /* Don't retry to open, something is permanently wrong.  */
+            this.isOpeningTerm = false;
+            this.isTermOpen = true;
+            return Promise.reject("No terminal");
+        }
+
+        /* This may have changed since we waited for waitForStarted. Test it again.  */
+        if (this.isVisible === false) {
+            this.isOpeningTerm = false;
+            return Promise.reject("Not visible");
+        }
+
+        this.term.open(this.node);
+        this.registerResize();
         this.connectSocket(this.terminalId);
         this.monitorTerminal(this.terminalId);
-    }
 
+        this.isTermOpen = true;
+        return Promise.resolve();
+    }
     protected createWebSocket(pid: string): WebSocket {
         const url = this.endpoint.getWebSocketUrl().resolve(pid);
         return this.webSocketConnectionProvider.createWebSocket(url.toString(), { reconnecting: false });
     }
 
-    protected onActivateRequest(msg: Message): void {
-        super.onActivateRequest(msg);
-        this.term.focus();
```

Comment:
Shouldn't the terminal still get focus on activation? I am thinking about a situation where I activate a terminal in another tab group which was shown before but just not active.

Reply:
Humm good point, I'll update it accordingly.

Reply:
Fixed in new version